packages/server/test/index.test.tsx

```diff
@@ -569,3 +571,85 @@ test('void mutation response', async () => {
   ws.close();
   close();
 });
+
+// https://github.com/trpc/trpc/issues/559
+describe('AbortError', () => {
+  test('cancelling request should throw AbortError', async () => {
+    const { client, close } = routerToServerAndClient(
+      trpc.router().query('slow', {
+        async resolve() {
+          await new Promise((resolve) => setTimeout(resolve, 500));
+          return null;
+        },
+      }),
+    );
+    const onReject = jest.fn();
+    const req = client.query('slow');
+    req.catch(onReject);
+    // cancel after 10ms
+    await new Promise((resolve) => setTimeout(resolve, 5));
+    req.cancel();
+
+    await waitFor(() => {
+      expect(onReject).toHaveBeenCalledTimes(1);
+    });
+
+    const err = onReject.mock.calls[0][0] as TRPCClientError<any>;
+
+    expect(err.name).toBe('TRPCClientError');
+    expect(err.originalError?.name).toBe('AbortError');
+
+    close();
+  });
+
+  test('cancelling batch request should throw TRPCAbortError', async () => {
+    // aborting _one_ batch request doesn't necessarily mean we cancel the reqs part of that batch
+    // therefore, we throw TRPCAbortError instead
```

Comment:
cc @kripod 

if you do a batch request it'll be `err.name === 'TRPCAbortError`

Reply:
do you think i should throw the same error for non-batch http requests too to ensure it's consistent?

Reply:
Yep, that sounds like a good idea – although I’m not sure why a `TRPCAbortError` is required at all. I think a `TRPCClientError` may be used over here too, wrapping `new DOMException("The operation was aborted.", "AbortError")` as its `originalError`.

Reply:
Great idea!